### PR TITLE
docs: add knative-rabbitmq.md

### PR DESCRIPTION
**Pull request by vincent0426**

### CODE REVIEW

File: docs/knative-rabbitmq.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+```sh
+kubectl rabbitmq manage rabbitmq-cluster
+```
+
+Then, navigate to [https://localhost:15672](https://localhost:15672/) and login with the credentials above.
+
+## Create Queue
+
+
+### Install cert-manager
```
Comment (dadiorchen) [inline docs/knative-rabbitmq.md:89]: Seems on greenstand k8s, we arleady have this

File: docs/knative-rabbitmq.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+```sh
+kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.11.0/serving-crds.yaml
+```
+
+```sh
+kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.11.0/serving-core.yaml
+```
+
+## Install network (kourier)
```
Comment (dadiorchen) [inline docs/knative-rabbitmq.md:118]: This means we don't use Istio?

File: docs/knative-rabbitmq.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+```
+
+Open another terminal then run
+
+```sh
+kn event send \
+  --to Broker:eventing.knative.dev/v1:broker
+```
+
+You should see our knative has created the pod to handle the trigger from event
```
Comment (dadiorchen) [inline docs/knative-rabbitmq.md:335]: Did you try the rabbitMQ plugin to: emit message from rabbitMQ and trigger knative call to a function?

**Comment by dadiorchen** [comment]
@vincent0426 good job!

**Comment by dadiorchen** [comment]
@vincent0426 how do you request your `hello world` function?

File: docs/knative-rabbitmq.md (hunk 2 above)
Comment (vincent0426) [reply docs/knative-rabbitmq.md:118]: I think we can still use Istio if prefered but I haven't try out that.

File: docs/knative-rabbitmq.md (hunk 3 above)
Comment (vincent0426) [reply docs/knative-rabbitmq.md:335]: I've tried rabbitmqadmin publish like `rabbitmqadmin publish exchange=amq.default routing_key=<trigger_name> payload="Hello rabbitmq" --username=<username> --password=<pwd>`

File: docs/knative-rabbitmq.md (hunk 2 above)
Comment (dadiorchen) [reply docs/knative-rabbitmq.md:118]: I'd like to leave this to you to decide, because you are the one most understand the case now.

**Comment by dadiorchen** [comment]
> @vincent0426 how do you request your `hello world` function?

Just in case you missed this, so, how do you use the function you deployed? How do we access it, test it

**Comment by dadiorchen** [comment]
@vincent0426 if you give you the cluster access on digitalocean, do you think you can help deploy all these things to cloud?

**Comment by vincent0426** [comment]
> > @vincent0426 how do you request your `hello world` function?
> 
> Just in case you missed this, so, how do you use the function you deployed? How do we access it, test it

We can use kn event to send a cloud event to broker, then it should send the req to trigger and trigger the function. If only test if the function is correct, I've added additional command.

**Comment by dadiorchen** [comment]
> and

I didn't get it, how do you send an event? By the knative command line (CLI)?

**Comment by vincent0426** [comment]
> > and
> 
> I didn't get it, how do you send an event? By the knative command line (CLI)?

Yes, under the Testing function section we can mock an event by using kn now

```
kn event send \
  --to Broker:eventing.knative.dev/v1:broker
```

**Comment by vincent0426** [comment]
> @vincent0426 I'm not fully understanding the event model, I have a question, currently, the event model is clear for the case of rabbitMQ, if a new message emerged, the event was triggered and then run the function in the knative, my question is how do we use the faas as a microservice? to expose the api/endpoint over http, so we can directly call/request the function and get the response?
> 
> This is another major use case and benefits for us to use faas, for some very small api, say, just one endpoint, it is too heavy to build a dedicated microservice app, so faas seems lightweight and quick to implement.

Hi @dadiorchen, I've added the example to curl url directly under testing section without using rabbitmq. Is this what you mean?

or you mean deploying a function to the service and connect to rabbitmq? I think this https://knative.dev/docs/getting-started/first-service/  in the yaml part can help. Basically, we need to create an image then push it to the registry and we can deploy it using knative service.

**Pull request closed without merging**